README
Uncolor. Aw jeah. Even used American spelling, because that's what
the original request used :D

Changelog
0.4:
   Added the recolor/recolor quick commands, with permissions (PEX). Now you can punch those
   wool blocks black and blue. Or green. Or orange. Up to you.
   
0.3: 
   Added colors to the on/off messages

0.2:
   Added a single-block quick mode, for when you just need to punch one block.

0.1:
   Initial version, just an on/off toggle for punching blocks so hard they turn white
 


src/net/turbogoat/UnColor/UnColorBlockListener.java
package net.turbogoat.UnColor;

import java.util.logging.Logger;

import org.bukkit.DyeColor;
import org.bukkit.block.Block;
import org.bukkit.entity.Player;
import org.bukkit.event.block.BlockDamageEvent;
import org.bukkit.event.block.BlockListener;


public class UnColorBlockListener extends BlockListener {
	private UnColor plugin;
	Logger log = Logger.getLogger("Minecraft");
	
	public UnColorBlockListener(UnColor instance) {
		plugin = instance;
	}
	
	public void onBlockDamage(BlockDamageEvent event) {
		Block b = event.getBlock();
		Player player = event.getPlayer();
		if (!player.getItemInHand().getType().toString().equals("AIR")) {
			return;
		}
		if(b.getType().toString().equalsIgnoreCase("WOOL")) {
			if (plugin.checkPlayerUCQuick(player) || plugin.checkPlayerUnColor(player)) {
				b.setData(DyeColor.WHITE.getData());
				return;
			}
			if (plugin.checkPlayerRCQuick(player) || plugin.checkPlayerReColor(player)) {
				this.reColorBlock(b, player);
			}
				

		}
	}
	
	private void reColorBlock(Block block, Player player) {
		Block b = block;
	    switch (plugin.getPlayerColor(player)) {
	    case 0:
	    	b.setData(DyeColor.WHITE.getData());
	    	break;
	    case 1:
	    	b.setData(DyeColor.ORANGE.getData());
	    	break;
	    case 2:
	    	b.setData(DyeColor.MAGENTA.getData());
	    	break;
	    case 3:
	    	b.setData(DyeColor.LIGHT_BLUE.getData());
	    	break;
	    case 4:
	    	b.setData(DyeColor.YELLOW.getData());
	    	break;
	    case 5:
	    	b.setData(DyeColor.LIME.getData());
	    	break;
	    case 6:
	    	b.setData(DyeColor.PINK.getData());
	    	break;
	    case 7:
	    	b.setData(DyeColor.GRAY.getData());
	    	break;
	    case 8:
	    	b.setData(DyeColor.SILVER.getData());
	    	break;
	    case 9:
	    	b.setData(DyeColor.CYAN.getData());
	    	break;
	    case 10:
	    	b.setData(DyeColor.PURPLE.getData());
	    	break;
	    case 11:
	    	b.setData(DyeColor.BLUE.getData());
	    	break;
	    case 12:
	    	b.setData(DyeColor.BROWN.getData());
	    	break;
	    case 13:
	    	b.setData(DyeColor.GREEN.getData());
	    	break;
	    case 14:
	    	b.setData(DyeColor.RED.getData());
	    	break;
	    case 15:
	    	b.setData(DyeColor.BLACK.getData());
	    	break;
	    }
	    return;
	}

}


src/net/turbogoat/UnColor/UnColor.java
package net.turbogoat.UnColor;

import java.util.HashMap;
import java.util.Map;
import java.util.logging.Logger;

import org.bukkit.ChatColor;
import org.bukkit.command.Command;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;
import org.bukkit.event.Event;
import org.bukkit.plugin.PluginManager;
import org.bukkit.plugin.java.JavaPlugin;

import ru.tehkode.permissions.PermissionManager;
import ru.tehkode.permissions.bukkit.PermissionsEx;

public class UnColor extends JavaPlugin {
	Logger log = Logger.getLogger("Minecraft");
	private final UnColorBlockListener blockListener = new UnColorBlockListener(this);
	public static Map<Player, Boolean> ucEnabled = new HashMap<Player, Boolean>();
	public static Map<Player, Boolean> ucqEnabled = new HashMap<Player, Boolean>();
	public static Map<Player, Boolean> rcEnabled = new HashMap<Player, Boolean>();
	public static Map<Player, Boolean> rcqEnabled = new HashMap<Player, Boolean>();
	public static Map<Player, Integer> rcColor = new HashMap<Player, Integer>();
	PermissionManager permissions = null;
		
	public void onEnable() {
		PluginManager pm = this.getServer().getPluginManager();
		pm.registerEvent(Event.Type.BLOCK_DAMAGE, blockListener, Event.Priority.Normal, this);
		log.info("UnColor plugin enabled.");
		if (this.getServer().getPluginManager().isPluginEnabled("PermissionsEx")) {
			permissions = PermissionsEx.getPermissionManager();
		}

	}
	public void onDisable() {
		log.info("UnColor plugin disabled.");
	}
	
	public int getPlayerColor(Player player) {
		if (rcColor.containsKey(player)) {
			return rcColor.get(player);
		}
		else {
			return 0;
		}
	}
	
	public boolean checkPlayerUCQuick(Player player) {
		if (ucqEnabled.containsKey(player)) {
			if(ucqEnabled.get(player)) {
				ucqEnabled.put(player, false);
				player.sendMessage(ChatColor.RED + "UnColor Quick-mode off");
				return true;
			}
		}
		return false;
	}
	
	public boolean checkPlayerUnColor(Player player) {
		if(ucEnabled.containsKey(player)) {
			if(ucEnabled.get(player)) {
					return true;
			} else {
				return false;
			}
		} else {
			ucEnabled.put(player, false);
			return false;
		}
	}
	
	public boolean checkPlayerReColor(Player player){
		if(rcEnabled.containsKey(player)) {
			return rcEnabled.get(player);
		}
		rcEnabled.put(player, false);
		return false;
	}
	
	public boolean checkPlayerRCQuick(Player player) {
		if (rcqEnabled.containsKey(player)) {
			if(rcqEnabled.get(player)) {
				rcqEnabled.put(player, false);
				player.sendMessage(ChatColor.RED + "ReColor Quick-mode off");
				return true;
			}
		}
		return false;
	}
	
	
	public boolean onCommand(CommandSender sender, Command cmd, String commandLabel, String[] args){
		if(cmd.getName().equalsIgnoreCase("uc")){
			Player player = (Player) sender;
			if (permissions != null) {
				if (!permissions.has(player, "uncolor.uncolor")) {
					player.sendMessage("You don't have permission to use this command");
					return true;
				}
			} 
			if (ucEnabled.containsKey(player)) {
				if (ucEnabled.get(player)) {
					ucEnabled.put(player, false);
					player.sendMessage(ChatColor.RED + "UnColor mode off.");
				} else {
					ucEnabled.put(player, true);
					player.sendMessage(ChatColor.GREEN + "UnColor mode on - Hit colored woolblocks to make " +
						"them white.");
				}
			} else {
				ucEnabled.put(player, true);
				player.sendMessage(ChatColor.GREEN + "UnColor mode on - Hit colored woolblocks to make " +
						"them white.");
			}
			ucqEnabled.put(player, false);
			rcEnabled.put(player, false);
			rcqEnabled.put(player, false);
			return true;
		}
		if(cmd.getName().equalsIgnoreCase("ucq")) {
			Player player = (Player) sender;
			if (permissions != null) {
				if (!permissions.has(player, "uncolor.uncolor")) {
					player.sendMessage("You don't have permission to use this command");
					return true;
				}
			}
			if (ucqEnabled.containsKey(player)){
				if (ucqEnabled.get(player)) {
					ucqEnabled.put(player, false);
					player.sendMessage(ChatColor.RED + "UnColor Quick-mode off");
				} else {
					ucqEnabled.put(player, true);
				    player.sendMessage(ChatColor.GREEN + "UnColor Quick-mode on - Hit a colored woolblock to make " +
							"it white.");
				}	
			} else {
				ucqEnabled.put(player, true);
				player.sendMessage(ChatColor.GREEN + "UnColor Quick-mode on - Hit a colored woolblock to make " +
						"it white.");
			}
			ucEnabled.put(player, false);
			rcEnabled.put(player, false);
			rcqEnabled.put(player, false);
			return true;
		}
		if(cmd.getName().equalsIgnoreCase("rc")) {
			Player player = (Player) sender;
			if (permissions != null) {
				if (!permissions.has(player, "uncolor.recolor")) {
					player.sendMessage("You don't have permission to use this command");
					return true;
				}
			}
			if (rcEnabled.containsKey(player)){
				if (rcEnabled.get(player) && args.length < 1) {
					player.sendMessage(ChatColor.RED + "ReColor off");
					rcEnabled.put(player, false);
				} else {
					if (args.length > 0) {
						if (this.setPlayerColor(player, args[0])) {
							player.sendMessage(ChatColor.GREEN + "Color set to: " + args[0]);
						} else {
							player.sendMessage(ChatColor.RED + "Invalid Color! use /rclist for a list of colors");
							return true;
						}
					}
					rcEnabled.put(player, true);
				    player.sendMessage(ChatColor.GREEN + "ReColor on - hit blocks to make them change color");
				}	
			} else {
				if (args.length > 0) {
					if (this.setPlayerColor(player, args[0])) {
						player.sendMessage(ChatColor.GREEN + "Color set to: " + args[0]);
					} else {
						player.sendMessage(ChatColor.RED + "Invalid Color! use /rclist for a list of colors");
						return true;
					}
				}
				player.sendMessage(ChatColor.GREEN + "ReColor on - hit blocks to make them change color");
				rcEnabled.put(player, true);
			}
			ucqEnabled.put(player, false);
			ucEnabled.put(player, false);
			rcqEnabled.put(player, false);
			return true;
			
		}
		if(cmd.getName().equalsIgnoreCase("rcq")) {
			Player player = (Player) sender;
			if (permissions != null) {
				if (!permissions.has(player, "uncolor.recolor")) {
					player.sendMessage("You don't have permission to use this command");
					return true;
				}
			}
			if (rcqEnabled.containsKey(player)){
				if (rcqEnabled.get(player) && args.length < 1) {
					player.sendMessage(ChatColor.RED + "ReColor Quick-Mode off");
					rcqEnabled.put(player, false);
				} else {
					if (args.length > 0) {
						if (this.setPlayerColor(player, args[0])) {
							player.sendMessage(ChatColor.GREEN + "Color set to: " + args[0]);
						} else {
							player.sendMessage(ChatColor.RED + "Invalid Color!");
							return true;
						}
					}
					rcqEnabled.put(player, true);
				    player.sendMessage(ChatColor.GREEN + "ReColor Quick-Mode on - hit a block to make it " +
				    	"change color");
				}	
			} else {
				if (args.length > 0) {
					if (this.setPlayerColor(player, args[0])) {
						player.sendMessage(ChatColor.GREEN + "Color set to: " + args[0]);
					} else {
						player.sendMessage(ChatColor.RED + "Invalid Color!");
						return true;
					}
				}
				player.sendMessage(ChatColor.GREEN + "ReColor Quick-Mode on - hit a block to make it " +
				    	"change color");
				rcqEnabled.put(player, true);
			}
			ucqEnabled.put(player, false);
			ucEnabled.put(player, false);
			rcEnabled.put(player, false);
			return true;
			
		}
		if(cmd.getName().equalsIgnoreCase("rclist")) {
			Player player = (Player) sender;
			if (permissions != null) {
				if (!permissions.has(player, "uncolor.recolor")) {
					player.sendMessage("You don't have permission to use this command");
					return true;
				}
			}
			player.sendMessage("Valid colors are: White, Black, Red, Magenta, Pink, Purple, Yellow, Orange, " +
				"Light_Blue, Blue, Gray, Light_Gray, Cyan, Brown, Green, Lime");
			return true;
		}
		return false;
	}
	
	private boolean setPlayerColor(Player player, String string) {
		if (string.equalsIgnoreCase("white")) {
			rcColor.put(player, 0);
			return true;
		}
		if (string.equalsIgnoreCase("orange")) {
			rcColor.put(player, 1);
			return true;
		}
		if (string.equalsIgnoreCase("magenta")) {
			rcColor.put(player, 2);
			return true;
		}
		if (string.equalsIgnoreCase("light_blue")) {
			rcColor.put(player, 3);
			return true;
		}
		if (string.equalsIgnoreCase("yellow")) {
			rcColor.put(player, 4);
			return true;
		}
		if (string.equalsIgnoreCase("lime")) {
			rcColor.put(player, 5);
			return true;
		}
		if (string.equalsIgnoreCase("pink")) {
			rcColor.put(player, 6);
			return true;
		}
		if (string.equalsIgnoreCase("gray")) {
			rcColor.put(player, 7);
			return true;
		}
		if (string.equalsIgnoreCase("light_gray")) {
			rcColor.put(player, 8);
			return true;
		}
		if (string.equalsIgnoreCase("cyan")) {
			rcColor.put(player, 9);
			return true;
		}
		if (string.equalsIgnoreCase("purple")) {
			rcColor.put(player, 10);
			return true;
		}
		if (string.equalsIgnoreCase("blue")) {
			rcColor.put(player, 11);
			return true;
		}
		if (string.equalsIgnoreCase("brown")) {
			rcColor.put(player, 12);
			return true;
		}
		if (string.equalsIgnoreCase("green")) {
			rcColor.put(player, 13);
			return true;
		}
		if (string.equalsIgnoreCase("red")) {
			rcColor.put(player, 14);
			return true;
		}
		if (string.equalsIgnoreCase("black")) {
			rcColor.put(player, 15);
			return true;
		}
		return false;
	}
}
